py3status/modules/netdata.py

```diff
@@ -18,14 +18,22 @@
 @author Shahin Azad <ishahinism at Gmail>
 """
 
+import os
 import subprocess
 
 
 class GetData:
     """Get system status.
     """
     def __init__(self, nic):
-        self.nic = nic
+        if nic is None:
+            iflist = os.listdir('/sys/class/net')
+            for n in iflist:
+                if n == 'lo':
+                    continue
+            self.nic = n
```

Comment:
This is broken you are just getting the last interface even if it is `lo`.

Also even if this was working it would wrong.  to keep existing behaviour if `eth0` is an interface then it should be selected.  If not then choosing any other one is fine.

Your selection is also naive and could be improved.  It could easily find a bridge rather than a physical network.  I'm also not sure if it should be preferring lan over wlan.

anyhow I'd check for `/sys/class/net/<dev>/device` for physical device and look in  `/sys/class/net/<dev>/uevent` for lan vs wlan.